flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/source/IcebergTableSource.java

```diff
@@ -175,6 +178,18 @@ public Result applyFilters(List<ResolvedExpression> flinkFilters) {
     return Result.of(acceptedFilters, flinkFilters);
   }
 
+  @Override
+  public void applySourceWatermark() {
+    if (!readableConfig.get(FlinkConfigOptions.TABLE_EXEC_ICEBERG_USE_FLIP27_SOURCE)) {
```

Comment:
Why is this not using `Preconditions` for the check?

Reply:
Thought it would be better to throw UnsupportedOperation Exception instead of IllegalArgument or IllegalStateException. If either of them is fine with  current error message , I can move to Preconditions.

Reply:
I would suggest to move to `Preconditions.checkArgument` - we usually stick to that for this kind of checks